Port libnotify support to 0.7

* Forwards/backwards-compatible, don't need a recompile when upgrading libnotify
* Haven't tested with older libnotify, but should work fine
* nn_new() takes the same no. of arguments because everything except the first
  argument is optional, and the last argument passed is supposed to be NULL.

---
--- a/src/fe-gtk/plugin-tray.c
+++ b/src/fe-gtk/plugin-tray.c
@@ -145,6 +145,7 @@
 libnotify_notify_new (const char *title, const char *text, GtkStatusIcon *icon)
 {
 	void *noti;
+	int libnotify_version = 0;
 
 	if (!nn_mod)
 	{
@@ -152,8 +153,13 @@
 		if (!nn_mod)
 		{
 			nn_mod = g_module_open ("libnotify.so.1", G_MODULE_BIND_LAZY);
-			if (!nn_mod)
-				return FALSE;
+			if (!nn_mod) {
+				nn_mod = g_module_open ("libnotify.so.4", G_MODULE_BIND_LAZY);
+				if (!nn_mod)
+					return FALSE;
+				libnotify_version = 4;
+			}
+			libnotify_version = 1;
 		}
 
 		if (!g_module_symbol (nn_mod, "notify_init", (gpointer)&nn_init))
@@ -161,7 +167,15 @@
 		if (!g_module_symbol (nn_mod, "notify_uninit", (gpointer)&nn_uninit))
 			goto bad;
 		if (!g_module_symbol (nn_mod, "notify_notification_new_with_status_icon", (gpointer)&nn_new_with_status_icon))
-			goto bad;
+			if (libnotify_version == 1)
+				/* We know we're using API version 1, but something is wrong */
+				goto bad;
+			else
+				/* It's probably API version 4, aka libnotify-0.7 */
+				libnotify_version = 4;
+		else
+			/* Only this API version has notify_notification_new_with_status_icon */
+			libnotify_version = 1;
 		if (!g_module_symbol (nn_mod, "notify_notification_new", (gpointer)&nn_new))
 			goto bad;
 		if (!g_module_symbol (nn_mod, "notify_notification_show", (gpointer)&nn_show))
